crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -102,6 +106,76 @@ impl Consensus {
         sha256::Hash::from_slice(leaf_hash.as_slice())
             .expect("parent_hash: Engines shouldn't be Err")
     }
+    /// Check whether a block is valid
+    /// All validation methods is outsorced to this function
+    ///
+    /// Checks:
+    ///     - Merkle root
+    ///     - BIP34
+    ///     - Witness commitment
+    ///     - Block size (inside Transactions validation)
+    ///     - Transactions
+    ///     - prev block hash
+    pub fn validate_block(
+        &self,
+        block: &bitcoin::Block,
+        ancestor: &BlockHeader,
+        height: u32,
+        inputs: HashMap<bitcoin::OutPoint, bitcoin::TxOut>,
+        flags: c_uint,
+        validate_script: bool,
+    ) -> Result<(), BlockchainError> {
```

Comment:
we still have the option to validate script or not by calling directly from any consensus object